### PR TITLE
Bugfix/hdev 1023 xsgd snapshot based on ave daily balance

### PR DESCRIPTION
**Pull request by schystz**
## Description of Changes

Script to generate daily rewards of LPs per epoch

Note: currently works for XSGD:USDC polygon pool only atm

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDEV-1023)

## How To Test

`yarn hardhat snapshot-xsgd-rewards --network polygon --startdate '2023-03'`

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
- [x] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally
- [x] I have formatted my code using format document in VSCode

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

**Review by tracyarciaga — APPROVED** [review]

File: scripts/util/halo-contract-address-network.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { mainnet, kovan, matic } from '@halodao/halodao-contract-addresses';
+
+export const haloContractAddresses = (network: string) => {
+  switch (network) {
+    case 'main':
```
Comment (0xAplki) [inline scripts/util/halo-contract-address-network.ts:5]: do we still need?

Comment (0xAplki) [inline scripts/util/halo-contract-address-network.ts:5]: do we still need?

File: scripts/util/halo-contract-address-network.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    case 'mainnet':
+      return mainnet;
+    case 'kovan':
+      return kovan;
+    case 'matic':
```
Comment (0xAplki) [inline scripts/util/halo-contract-address-network.ts:10]: do we still need?

**Review by 0xAplki — APPROVED** [review]
minor comment but generally looks good!

**Pull request merged**